### PR TITLE
Bug fix in deep_eq when comparing arrays

### PR DESCRIPTION
**Pull request by barkm**
Fixes a bug in `Config::deep_eq` when comparing arrays.

It would probably be good to add a unit test to cover this, but I'm not sure how to do that.

### CODE REVIEW

File: configuru.hpp [hunk 1]
```diff
@@ -1787,7 +1787,7 @@ namespace configuru
 			auto&& b_array = b.as_array();
 			if (a_array.size() != b_array.size()) { return false; }
 			for (size_t i=0; i<a_array.size(); ++i) {
-				if (!deep_eq(a_array[i], a_array[i])) {
+				if (!deep_eq(a_array[i], b_array[i])) {
```
Comment (bylowerik) [inline configuru.hpp:1790]: 👍

**Review by bylowerik — APPROVED** [review]
LGTM

**Comment by vidstige** [comment]
I think you can probably add it [here](https://github.com/emilk/Configuru/blob/master/test%2Fconfiguru_test.cpp). But :man_shrugging: :-)

**Review by emilk — APPROVED** [review]
Good catch ❤️

**Pull request merged**